srml/balances/src/lib.rs

```diff
@@ -335,7 +335,9 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 
 	/// Kill an account's free portion.
 	fn on_free_too_low(who: &T::AccountId) {
-		Self::decrease_total_stake_by(Self::free_balance(who));
+		// underflow should never happen, but if it does, there's not much we can do about it.
+		let _ = Self::decrease_total_stake_by(Self::free_balance(who));
```

Comment:
Maybe we could introduce a set of Events for this particular cases, like standard event in system, `Underflow(String)` and `Overflow(String)` with String being the name of the value ?

Reply:
First, you most likely don't want to use `String`, since we strive to avoid it in the runtime logic. 
Second, what use-cases do you see for this?

Reply:
The idea is that it is better to be aware that a saturated overflow or underflow happens than it being silent

Reply:
if overflow/underflow happens, then it's a fundamental logic error/state corruption and governance or a hard fork is needed to fix it. it'll be obvious if it happens.